static/app/utils/tokenizeSearch.tsx

```diff
@@ -36,6 +39,37 @@ function isParen(token: Token, character: '(' | ')') {
 export class MutableSearch {
   tokens: Token[];
 
+  /**
+   * Creates a `MutableSearch` from a key-value mapping of field:value.
+   * This construct doesn't support conditions like `OR` and `AND` or
+   * parentheses, so it's only useful for simple queries.
+   * @param params
+   * @returns {MutableSearch}
+   */
+  static fromQueryObject(params: {
```

Comment:
l: I dislike static methods on classes when writing browser JavaScript because

a) unable to get treeshaken, always included if `MutableSearch` is used in anyway in the bundle
b) `fromQueryObject` is [unminifiable](https://blog.sentry.io/performance-impact-of-generated-javascript/#converting-classes-to-objects-and-functions-and-minimizing-private-fields) as JS needs the full-name to access the method on the class, unnecessary bundle size. `MutableSearch.fromQueryObject` gets minified to `MutableSearch.fromQueryObject` 😢

I much prefer `createMutableSearchFromQueryObject` function as an alternative, especially because we already namespace by file that it is being exported from.

In this case I don't think it's a huge deal because `fromQueryObject` will be pretty commonly used, and it's not gonna be a huge part of the bundle considering all the code, but I prefer not encouraging the pattern in the codebase.

Reply:
🤔 minification is an interesting point! On the other hand, like you said, it's peanuts in terms of bytes, I think the Sentry codebase isn't nearly as sensitive to this as the SDK, right?

Static methods as factory methods is _such_ a common software pattern, I'd hate to break it here (see `EventView`, `Flamegraph`, etc.) though I'll admit I'm _mostly_ inspired by Python and Ruby patterns on this one

Helper methods like `createMutableSearchFromQueryObject` has discoverability problems, they tend to drift away from underlying implementations, and so on and like that

Reply:
> On the other hand, like you said, it's peanuts in terms of bytes, I think the Sentry codebase isn't nearly as sensitive to this as the SDK, right

For this specific case it's not, but in aggregate this does add up as this pattern is used more and more! Especially because every individual call to the static method is unminified.

> Helper methods like createMutableSearchFromQueryObject has discoverability problems, they tend to drift away from underlying implementations, and so on and like that

I disagree with this because the function still returns an object instance with `MutableSearch`, and changes in output can be detected with tests like you've added. One can argue that any method (like hooks) that rely on entities like `project`, `organization` suffer from the same possible drifting issues.

I don't want to unblock, feel free to merge though!! Thanks for responding to all my opinions!

Reply:
I'm happy to get the feedback/questions!